internal/version/version.go

```diff
@@ -28,16 +36,44 @@ var (
 )
 
 func init() {
-	// This regexp matches the version format we use and captures major/minor/patch/rc in different groups
-	r := regexp.MustCompile(`v(?P<ma>\d+)\.(?P<mi>\d+)\.(?P<pa>\d+)(-rc\.(?P<rc>\d+))?`)
-	names := r.SubexpNames()
-	captures := map[string]string{}
-	// Associate each capture group match with the capture group's name to easily retrieve major/minor/patch/rc
-	for k, v := range r.FindAllStringSubmatch(Tag, -1)[0] {
-		captures[names[k]] = v
+	// Check if we are using a transitional v1.74.x or later version
+	info, _ := debug.ReadBuildInfo()
+	var v1Version string
+	for _, dep := range info.Deps {
+		if dep.Path == "gopkg.in/DataDog/dd-trace-go.v1" {
+			v1Version = dep.Version
+			break
+		}
 	}
-	Major, _ = strconv.Atoi(captures["ma"])
-	Minor, _ = strconv.Atoi(captures["mi"])
-	Patch, _ = strconv.Atoi(captures["pa"])
-	RC, _ = strconv.Atoi(captures["rc"])
+	if v1Version != "" {
+		major, minor, _, _ := parseVersion(v1Version)
+		if major == 1 && minor >= 74 {
+			V1Tag.Transitional = true
+			V1Tag.Version = v1Version
+			Tag = v1Version
+		}
+	}
+	Major, Minor, Patch, RC = parseVersion(Tag)
+}
+
+func parseVersion(version string) (int, int, int, int) {
+	var (
+		v     = semver.MustParse(version)
```

Comment:
I decided to go with `MustParse` because this shouldn't fail ever. I consider this a canary.